React component refs

KerimRahman opened the pull request.

KerimRahman committed:
refactor(react/Button): forward ref

KerimRahman committed:
refactor(react/Modal): forward ref

kidroca commented on packages/react/src/components/Button/Button.tsx:
```diff
@@ -11,14 +11,19 @@ export interface ButtonProps {
 }
 
-const Button = ({ label, mode = 'primary', size = 'medium', disabled = false, className, style, onClick }: ButtonProps) => {
+export type Ref = HTMLButtonElement;
+
+const Button = forwardRef<Ref, ButtonProps>((props, ref) => {
```
Why do we `export type Ref` ? 

vs just
```jsx
const Button = forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
```

kidroca commented on packages/react/src/components/Modal/Modal.tsx:
```diff
@@ -12,18 +12,23 @@ export interface ModalProps {
 }
 
-const Modal = ({ title = '', size = 'default', onClose, footerSlot, className, style, children }: ModalProps) => {
+export type Ref = HTMLDivElement;
```
we should avoid having exports with the same name for easier usage, 
if we have to export this, prefer `ModalRef` or something similar

KerimRahman committed:
refactor(react/components): code refactor

kidroca committed:
refactor(packages/react): update component props interfaces for ...rest

Add type annotations to show available html attributes in IDE autocomplete

kidroca committed:
chore(package/react): update storybook docs propFilter

Allow some native attributes added from extended interfaces to appear in
props documentation

kidroca committed:
style(package/react): reformat with prettier
